### PR TITLE
feat(radon): implement average mean for RadonArray

### PR DESCRIPTION
**Pull request by lrubiorod**

### CODE REVIEW

File: rad/src/operators/array.rs [hunk 1]
```diff
@@ -58,6 +58,36 @@ pub fn map(input: &RadonArray, args: &[Value]) -> Result<RadonTypes, RadError> {
...
+
+                if prev_len.is_none() {
+                    for sub_item in rad_value.value().into_iter() {
+                        v.push(vec![sub_item]);
+                    }
+                    prev_len = Some(sub_value_len);
+                } else if prev_len == Some(sub_value_len) {
+                    for (i, sub_item) in rad_value.value().into_iter().enumerate() {
+                        let v_aux = v.get_mut(i).unwrap();
+                        v_aux.push(sub_item);
```
Comment (tmpolaczyk) [inline rad/src/operators/array.rs]:
```suggestion
                        v[i].push(sub_item);
```

File: rad/src/operators/array.rs [hunk 2]
```diff
@@ -58,6 +58,36 @@ pub fn map(input: &RadonArray, args: &[Value]) -> Result<RadonTypes, RadError> {
...
+    let mut prev_len = None;
+    for item in input.value() {
+        match item {
+            RadonTypes::Array(rad_value) => {
+                let sub_value = rad_value.value();
+                let sub_value_len = sub_value.len();
+
+                if prev_len.is_none() {
+                    for sub_item in rad_value.value().into_iter() {
+                        v.push(vec![sub_item]);
```
Comment (tmpolaczyk) [inline rad/src/operators/array.rs]: You could optimize this by preallocating a vector with capacity for `input.value().len()` elements, but it's a very small optimization.

File: rad/src/operators/array.rs [hunk 3]
```diff
@@ -58,6 +58,36 @@ pub fn map(input: &RadonArray, args: &[Value]) -> Result<RadonTypes, RadError> {
     Ok(RadonArray::from(result).into())
 }
 
+pub fn transpose(input: &RadonArray) -> Result<Vec<Vec<RadonTypes>>, RadError> {
```
Comment (tmpolaczyk) [inline rad/src/operators/array.rs]: If the input is `&RadonArray`, shouldn't the output be `Result<RadonArray, RadError>`?

Comment (tmpolaczyk) [reply rad/src/operators/array.rs]: And add tests please, especially for the error cases

File: rad/src/reducers/average.rs [hunk 4]
```diff
@@ -1,24 +1,167 @@
...
+    // Float case
+    if inner_type == discriminant(&RadonTypes::Float(RadonFloat::default())) {
+        let sum = value.iter().try_fold(0f64, |sum, item| match item {
+            RadonTypes::Float(f64_value) => Ok(sum + f64_value.value()),
+            _ => Err(RadError::MismatchingTypes),
         });
+        let sum = sum?;
+
+        // Divide sum by the count of numeric values that were summed
+        let mean_value: f64 = sum.div(value_len as f64);
```
Comment (tmpolaczyk) [inline rad/src/reducers/average.rs]: What should happen if `value_len == 0`?

Currently, it returns `NaN`. Add test:

```rust
#[test]
fn test_operate_reduce_mean_empty() {
    let input = RadonArray::from(vec![]);
    let output = mean(&input).unwrap();
    assert_eq!(
        output,
        RadonTypes::from(RadonFloat::from(std::f64::NAN))
    );
}
```

File: rad/src/error.rs [hunk 5]
```diff
@@ -117,6 +117,12 @@ pub enum RadError {
     /// Overflow error
     #[fail(display = "Overflow error")]
     Overflow,
+    /// Mismatching types
+    #[fail(display = "Mismatching types")]
```
Comment (aesedepece) [inline rad/src/error.rs]: This error message is lacking context. We should print both values for debugging purposes.

File: rad/src/error.rs [hunk 6]
```diff
@@ -117,6 +117,12 @@ pub enum RadError {
...
+    MismatchingTypes,
+    /// Arrays to be reduced have different sizes
+    #[fail(display = "Arrays to be reduced have different sizes")]
```
Comment (aesedepece) [inline rad/src/error.rs]: - For context, e should explicitly state which reducer thrown the error.
- We can print the sizes here as `(x != y)`

File: rad/src/types/array.rs [hunk 7]
```diff
@@ -31,6 +34,26 @@ impl RadonArray {
         self.inner_type
     }
 
+    pub fn inner_type_name(&self) -> String {
```
Comment (aesedepece) [inline rad/src/types/array.rs]: Wow this is wild. There's no reason for calling `self.inner_type()` all the time.

**Pull request merged**